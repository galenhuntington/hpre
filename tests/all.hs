{-# LANGUAGE LambdaCase, UnicodeSyntax #-}

import Data.List as List
import qualified Control.Monad as Monad

--+
import Data.Char
import Data.String as String

str1 :: String
''  = "test1"

str2 :: String
〃  = "test2"

str3 :: String
str3 = str4 where
   str4 :: String
   '' = "test3"

isPos :: Int -> Bool
''  x | x > 0 = True
      |       = False

isPos' :: Int -> Bool
''  = \case
   x | x > 0 -> True
     |       -> False

isPosU ∷ Int → Bool
”   = \case
   x | x > 0 → True
     |       → False

list :: [Int]
''  = [
   1,
   2,
   ]

list' :: [Int]
''  = [
   , 1
   , 2
   ]

data Test =
   | Test1
   | Test2

tabbed :: Int
''  = x where
	x = 1

number :: Int
number = 299_792_458.123_456

