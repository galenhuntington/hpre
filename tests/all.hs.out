{-# LANGUAGE LambdaCase, UnicodeSyntax #-}

import Data.List as List
import qualified Control.Monad as Monad


import Data.Char 
import qualified Data.String  as String

str1 :: String
str1{-#COLUMN 5#-}= "test1"

str2 :: String
str2{-#COLUMN 4#-}= "test2"


isPos :: Int -> Bool
isPos{-#COLUMN 5#-}x | x > 0 = True
      | True  = False

isPos' :: Int -> Bool
isPos'{-#COLUMN 5#-}= \case
   x | x > 0 -> True
     | True  -> False

isPosU ∷ Int → Bool
isPosU{-#COLUMN 5#-}= \case
   x | x > 0 → True
     | True  → False

list :: [Int]
list{-#COLUMN 5#-}= [
   1,
   2
   ]

list' :: [Int]
list'{-#COLUMN 5#-}= [
     1
   , 2
   ]

data Test =
     Test1
   | Test2

tabbed :: Int
tabbed{-#COLUMN 5#-}= x where
   x = 1

number :: Int
number = 299792458.123456

